src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -24,6 +24,42 @@ internal TagHelperAttributeDescriptor([NotNull] string name, [NotNull] PropertyI
         }
 
         /// <summary>
+        ///  Internal for testing.
+        /// </summary>
+        internal TagHelperAttributeDescriptor(
+            [NotNull] string name,
+            [NotNull] string propertyName,
+            [NotNull] string typeName,
+            bool isIndexer)
+            : this(
+                name,
+                propertyName,
+                typeName,
+                isIndexer,
+                isStringProperty: string.Equals(typeName, typeof(string).FullName, StringComparison.Ordinal))
+        {
+        }
+
+        /// <summary>
+        ///  Internal for testing.
+        /// </summary>
+        internal TagHelperAttributeDescriptor(
+            [NotNull] string name,
+            [NotNull] string propertyName,
+            [NotNull] string typeName,
+            bool isIndexer,
+            bool isStringProperty)
+            : this(
+                  name,
+                  propertyName,
+                  typeName,
+                  isIndexer,
+                  isStringProperty: string.Equals(typeName, typeof(string).FullName, StringComparison.Ordinal),
+                  useage: null)
+        {
+        }
```

Comment:
In real PR, update tests to use more-complete constructors and remove these. Same in `TagHelperDescriptor`.